command/license_put_test.go

```diff
@@ -0,0 +1,28 @@
+package command
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+)
+
+var _ cli.Command = &LicensePutCommand{}
+
+func TestCommand_LicensePut_OSSErr(t *testing.T) {
+	t.Parallel()
+
+	srv, _, url := testServer(t, false, nil)
+	defer srv.Shutdown()
+
+	ui := new(cli.MockUi)
+	cmd := &LicensePutCommand{Meta: Meta{Ui: ui}, testStdin: strings.NewReader("testlicenseblob")}
+
+	if code := cmd.Run([]string{"-address=" + url, "-"}); code != 1 {
+		require.Equal(t, code, 1)
+	}
+
+	require.Contains(t, ui.ErrorWriter.String(), "Nomad Enterprise only endpoint")
```

Comment:
Would be nice if the test can be the same whether it's OSS or Enterprise.  We've had this issue in other tests too.

I would suggest having `testServer` `TestAgent` indicate if it's Enterprise, and use that for assertions.  It can be backed by a compile time constant guarded by `ent` compile tag.